devdoc/WorkUnits.rst

```diff
@@ -430,12 +430,47 @@ Once all of the required workflow items are executed, the workunit is marked as
 workflow items that are waiting to be triggered by an event, the workunit will be passed to the scheduler,
 which will keep monitoring for events.
 
+Note that most items in the xml workflow begin in the state WFStateNull. This means that it is valid to execute them, but they haven't been executed yet. Typically, only a few items begin in the state required.
+
 There are various specialised types of workflow items - e.g. sequential, wait, independent, but they all follow the same basic approach of
 executing dependencies and then executing that particular item.
 
 Most of the interesting work in an ECL query is done within a graph.  The call ctx->executeGraph will either execute the graph locally
 (in the case of hthor and roxie), or add the workunit onto a queue (for Thor).  Whichever happens, control will pass to that engine.
 
+Specialised Workflow Items
+=================
+Each item mode/type can affect the dependency structure of the workflow:
+
+- Sequential/Ordered
+  An item is made to contain all of the actions in the sequential/ordered statement. Each action becomes a dependency of this item (unless there is no item for the action). The dependencies, and therefore the actions, must be executed in order.
+  For sequential, an extra item is always inserted between the sequential item and each of its dependencies. This means that if other sequential statements reference the same attribute, it will only be performed once.
```

Comment:
For both I think? Not sure on the best wording.  If one of the child action is purely another workflow item, an extra item is still created to represent the action separately from the dependency.